tensorflow_datasets/image/lost_and_found.py

```diff
@@ -0,0 +1,223 @@
+          description='Lost and Found stereo images and disparity maps.',
+          version="1.0.0",
+          right_images=True,
+          segmentation_labels=False,
+          instance_ids=False,
+          disparity_maps=True,
+          use_16bit=False,
+      ),
+      LostAndFoundConfig(
+          name='full',
+          description='Full Lost and Found dataset.',
+          version="1.0.0",
+          right_images=True,
+          segmentation_labels=True,
+          instance_ids=True,
+          disparity_maps=True,
+          use_16bit=False,
+      ),
+      LostAndFoundConfig(
+          name='full_16bit',
+          description='Full Lost and Found dataset.',
+          version="1.0.0",
+          right_images=True,
+          segmentation_labels=True,
+          instance_ids=True,
+          disparity_maps=True,
+          use_16bit=True,
+      )]
+
+  def _info(self):
+    possible_features = {
+        'image_left': tfds.features.Image(shape=(1024, 2048, 3),
+                                          encoding_format='png'),
+        'image_right': tfds.features.Image(shape=(1024, 2048, 3),
+                                           encoding_format='png'),
+        'segmentation_label': tfds.features.Image(shape=(1024, 2048, 1),
+                                                  encoding_format='png'),
+        'instance_id': tfds.features.Image(shape=(1024, 2048, 1),
+                                           encoding_format='png'),
+        'disparity_map': tfds.features.Image(shape=(1024, 2048, 1),
+                                             encoding_format='png')}
+    return tfds.core.DatasetInfo(
+        builder=self,
+        # This is the description that will appear on the datasets page.
+        description=_DESCRIPTION,
+        # tfds.features.FeatureConnectors
+        features=tfds.features.FeaturesDict({
+            'image_id': tfds.features.Text(),
+            **{feat: possible_features[feat]
+               for feat in self.builder_config.features}}),
+        # Homepage of the dataset for documentation
+        urls=['http://www.6d-vision.com/lostandfounddataset'],
+        citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    """Returns SplitGenerators."""
+    base_url = 'http://www.dhbw-stuttgart.de/~sgehrig/lostAndFoundDataset/{}.zip'
+    download_urls = {
+        'image_left': base_url.format(self.builder_config.left_image_string)}
+    if 'image_right' in self.builder_config.features:
+      download_urls['image_right'] = base_url.format(
+          self.builder_config.right_image_string)
+    if 'segmentation_label' in self.builder_config.features \
+            or 'instance_id' in self.builder_config.features:
+      download_urls['gt'] = base_url.format('gtCoarse')
+    if 'disparity_map' in self.builder_config.features:
+      download_urls['disparity_map'] = base_url.format('disparity')
+    # split into two steps to save space for testing data
```

Comment:
I didn't understand this part. Can you explain more why you decided not to use download_and_extract ?

Reply:
If I run `download_and_extract` the test will fail because it apparently skips the whole command during testing, instead of extracting those zip-files specified in `DL_EXTRACT_RESULTS` in `lost_and_found_test.py`.

Splitting the two steps correctly replaces the results from the download-function with `DL_EXTRACT_RESULTS` and subsequently extracts the archives in the unittest, which means that the test-data can be stored in compressed zip-archives, which saves space.

I will update the comment to explain this better.
Or is there another workaround to fix this issue?